src/extension/IOController_Posix.hpp

```diff
@@ -127,6 +128,13 @@ namespace extension {
                 task.processing_events = task.waiting_events;
                 task.waiting_events    = 0;
 
+                // Mask out the currently processing events so poll doesn't notify for them
+                for (auto& w : watches) {
+                    if (task.fd == w.fd) {
+                        w.events = event_t(w.events & ~task.processing_events);
+                    }
+                }
```

Comment:
Can binary search (also I think those that call fire_event already has the watch found?)

Reply:
Does it?